plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -93,25 +95,66 @@
 {
     private static final Logger log = Logger.get(BaseJdbcClient.class);
 
+    public static final Type TRINO_PAGE_SINK_ID_COLUMN_TYPE = BigintType.BIGINT;
+
+    public static String getPageSinkIdColumnName(List<String> otherColumnNames)
+    {
+        // While it's unlikely this column name will collide with client table columns,
+        // guarantee it will not by appending a deterministic suffix to it.
+        String baseColumnName = "trino_page_sink_id";
+        String columnName = baseColumnName;
+        int suffix = 1;
+        while (otherColumnNames.contains(columnName)) {
+            columnName = baseColumnName + "_" + suffix;
+            suffix++;
+        }
+        return columnName;
+    }
+
+    public static ColumnMetadata getPageSinkIdColumn(List<String> otherColumnNames)
+    {
+        return new ColumnMetadata(getPageSinkIdColumnName(otherColumnNames), TRINO_PAGE_SINK_ID_COLUMN_TYPE);
+    }
+
     protected final ConnectionFactory connectionFactory;
     protected final QueryBuilder queryBuilder;
     protected final String identifierQuote;
     protected final Set<String> jdbcTypesMappedToVarchar;
     private final IdentifierMapping identifierMapping;
 
+    private final boolean supportsFTE;
```

Comment:
do not abbreviate. Use either `supportsRetries` or `supportsFaultTolerantExecution`. I would go with first one.